src/systems/implicit_system.C

```diff
@@ -113,6 +107,11 @@ void ImplicitSystem::init_matrices ()
   if (matrix->initialized())
     return;
 
+  // Clear any existing matrices
```

Comment:
I am not sure if I should move this code inside here. Let me try move it back. I will need to run MOOSE test to make sure. BTW, how do I run libmesh test locally?